### PR TITLE
Update the payment-request IDL file

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

**Review by marcoscaceres — APPROVED** [review]
Thanks for updating. This is great. More updates hopefully landing this week - though I think they are just dictionaries.

File: payment-request/idlharness.https.window.js [hunk 1]
```diff
@@ -10,16 +10,21 @@ idl_test(
...
+      const details = {total: {label: 'label', amount: amount} };
       window.paymentRequest = new PaymentRequest(methods, details);
     } catch (e) {
-      // Will be surfaced in idlharness.js's test_object below.
+      // Surfaced below when paymentRequest is undefined.
     }
 
     idlArray.add_objects({
       PaymentRequest: ['paymentRequest'],
+      PaymentMethodChangeEvent: ['new PaymentMethodChangeEvent("type")'],
```
Comment (marcoscaceres) [inline payment-request/idlharness.https.window.js]: I guess the "type" doesn't really matter, right?

Comment (marcoscaceres) [reply payment-request/idlharness.https.window.js]: (if it does, "paymentmethodchange" would be the right type here)

Comment (lukebjerring) [reply payment-request/idlharness.https.window.js]: I don't think it would cause an issues, but doesn't hurt to use real values.

**Comment by lukebjerring** [comment]
np - I actually took a stab at adding a `PaymentResponse` object too, until I noticed there's no way to automate the user's input of the payment (all the other tests are manual).. let me know if you make progress on that front. I think @kereliuk will have some authority on what's possible in terms of webdriver stuffs.

**Comment by lukebjerring** [comment]
@marcoscaceres - my general (limited) understanding is that, prior to a standardized webdriver API/implementation, we'd want some `testdriver.js` methods which trigger the user-controlled flow of the payment request (cancel, approve, add a card... , idk what else ). That `testdriver.js` method's default behaviour is 'reject as unimplemented'.

Then, each browser will write its `testdriver-vendor.js` specific implementation of said API, which isn't maintained it WPT, and may only work for that browser when it's in a test build. We benefit from sharing the tests, even if they all fail when being run in WPT.

https://github.com/web-platform-tests/wpt/blob/master/docs/_writing-tests/testdriver-tutorial.md

So, knowing very little about payment-requests - is there any (payment-request specific) automation behaviour available in any of the browsers?

**Comment by marcoscaceres** [comment]
> So, knowing very little about payment-requests - is there any (payment-request specific) automation behaviour available in any of the browsers?

No, there is none. It's too hard to automate. Best we can do is these clicks.

**Pull request merged**